0.4/tests/test_validation.py

```diff
@@ -3,39 +3,53 @@
 
 import pytest
 
-from jsonschema import validate
+from jsonschema import RefResolver, Draft202012Validator
 from jsonschema.exceptions import ValidationError
 
 
-def files():
-    return list(glob.glob(f"examples/valid/*.json")) + \
-        list(glob.glob(f"examples/valid_if_not_strict/*.json")) + \
-        list(glob.glob(f"examples/invalid/*.json"))
+with open('schemas/image.schema') as f:
+    image_schema = json.load(f)
+with open('schemas/strict_image.schema') as f:
+    strict_image_schema = json.load(f)
+schema_store = {
+    image_schema['$id']: image_schema,
+    strict_image_schema['$id']: strict_image_schema,
+}
 
-def ids():
-    return [str(x).split("/")[-1][0:-5] for x in files()]
+resolver = RefResolver.from_schema(image_schema, store=schema_store)
+validator = Draft202012Validator(image_schema, resolver=resolver)
+strict_validator = Draft202012Validator(strict_image_schema, resolver=resolver)
 
+valid_strict_files = list(glob.glob("examples/valid_strict/*.json"))
+valid_files = list(glob.glob("examples/valid/*.json"))
+invalid_files = list(glob.glob("examples/invalid/*.json"))
 
-@pytest.mark.parametrize("testfile", files(), ids=ids())
-def test_json(testfile):
 
-    if "examples/invalid/" in testfile:
+def ids(files):
+    return [str(x).split("/")[-1][0:-5] for x in files]
+
+
+@pytest.mark.parametrize(
+    "testfile", valid_strict_files, ids=ids(valid_strict_files))
+def test_valid_strict(testfile):
+    with open(testfile) as f:
+        json_file = json.load(f)
+        validator.validate(json_file)
+        strict_validator.validate(json_file)
+
+
+@pytest.mark.parametrize("testfile", valid_files, ids=ids(valid_files))
+def test_valid_files(testfile):
+    with open(testfile) as f:
+        json_file = json.load(f)
+        validator.validate(json_file)
+        with pytest.raises(ValidationError):
+            strict_validator.validate(json_file)
+
+
+@pytest.mark.parametrize("testfile", invalid_files, ids=ids(invalid_files))
+def test_invalid(testfile):
+    with open(testfile) as f:
+        json_file = json.load(f)
         with pytest.raises(ValidationError):
-            json_schema(testfile)
-    else:
-        json_schema(testfile)
-
-
-def json_schema(path):
-    # Load the correct schema
-    with open(path) as f:
-        test_json = json.loads(f.read())
-        # we don't have @type in this version
-        if "multiscales" in test_json:
-            schema_name = "image.schema"
-        else:
-            raise Exception("No schema found")
-
-    with open('schemas/' + schema_name) as f:
-        schema = json.loads(f.read())
-    validate(instance=test_json, schema=schema)
+            validator.validate(json_file)
```

Comment:
Is it worth confirming that these 'invalid' files also fail with the `strict_validator`?

Reply:
Yes I debated that. I ended up assuming it's a given since the strict schema is an `allOf` but adding the extra tests cannot hurt

Reply:
Fixed in [f0f3930](https://github.com/ome/ngff/pull/92/commits/f0f3930cd2ee841fbb8ed0f5c341e6fcd4110d63)